### ISSUE TITLE
webui allows to create self referencing repositories

### ISSUE THREAD

**Issue by adrianschroeter** [openSUSE/open-build-service#2362]
webui/project#destroy threw this 8 times
<pre>
SystemStackError: stack level too deep
</pre>
[https://build.opensuse.org/project/destroy](https://build.opensuse.org/project/destroy)"

[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/57fe2f2f954b9a0006000000)

**Comment by adrianschroeter** [comment]
We used to check for self references in releases before 2.6 .

While the backend can handle cycles we should not allow to create them, esp since we break in lots of places like this example.

**Comment by saraycp** [comment]
We wonder if it is still valid. `2.7` is not maintained any longer and trying to reproduce it in Unstable, it seems the error is not happening.

Are we trying it correctly? We did:
- added a Repository to the project
- added a Repository Path pointing to the previously created one
- deleted the project
- checked there weren't any errors

![Screenshot_2019-11-27 Open Build Service(4)](https://user-images.githubusercontent.com/2581944/69742957-e6bc5180-113d-11ea-9587-321f7dfb8a98.png)


In case we followed the right steps, I would say we can close this issue.
